packages/tasit-action/src/TasitAction.js

```diff
@@ -301,27 +278,45 @@ class ProviderFactory {
       throw new Error(`Invalid provider, use: [${providers}].`);
     }
 
+    if (provider === "fallback") network = "default";
     if (network === "mainnet") network = "homestead";
     else if (network === "other") network = undefined;
 
+    const defaultConfig = ProviderFactory.getDefaultConfig();
+
+    let ethersProvider;
+
     switch (provider) {
       case "default":
-        return ethers.getDefaultProvider(network);
+        ethersProvider = ethers.getDefaultProvider(network);
+
       case "infura":
-        return new ethers.providers.InfuraProvider(
+        ethersProvider = new ethers.providers.InfuraProvider(
           network,
-          infura.apiAccessToken
+          infura.apiKey
         );
+
       case "etherscan":
-        return new ethers.providers.EtherscanProvider(
+        ethersProvider = new ethers.providers.EtherscanProvider(
           network,
-          ethescan.apiToken
+          etherscan.apiKey
         );
+
       case "jsonrpc":
-        let p = new ethers.providers.JsonRpcProvider(jsonRpc, network);
-        if (pollingInterval) p.pollingInterval = pollingInterval;
-        return p;
+        let { url, port, user, password, allowInsecure } = jsonRpc;
+        if (url === undefined) url = defaultConfig.jsonRpc.url;
```

Comment:
Maybe log something in this case to say "using default url"?

Reply:
Thread extract to issue: https://github.com/tasitlabs/TasitSDK/issues/103

Reply:
Thanks! I'm resolving this.